Makefile

```diff
@@ -94,6 +99,9 @@ build-adapter: generate-api pkg/scalers/liiklus/LiiklusService.pb.go
 generate-api:
 	$(GO_BUILD_VARS) operator-sdk generate k8s
 	$(GO_BUILD_VARS) operator-sdk generate crds
+	# withTriggers is only used for duck typing so we only need the deepcopy methods
+	# However operator-sdk generate doesn't appear to have an option for that
+	rm deploy/crds/keda.sh_withtriggers_crd.yaml
```

Comment:
@zroubalik I'm not sure if you know a better way of doing this

Reply:
There should be a marker to disable the generation. I am not sure. Let me find it and fix it. I guess we can do it like that in the meantime

Reply:
There is a marker to disable CRD generation for all types in a package (so we would need to move this type to another api eg. `apis/duck/`. But this would cause problems for the ducktype itself, because `WithTriggers` and `ScaleTriggers` would be in different packages.
There is opened issue to add a marker to disable crd generation for a type. This should solve our problem. I think we can keep this workaround till the issue is fixed upstream.
https://github.com/kubernetes-sigs/controller-tools/issues/398